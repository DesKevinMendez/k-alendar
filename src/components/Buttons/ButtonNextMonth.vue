<template>
  <button type="button" class="k-alendar-navegation-left">&rarr;</button>
</template>

<script setup lang="ts"></script>


<style scoped lang="postcss">
.k-alendar-navegation-left {
    @apply px-2 py-1 border border-gray-200 rounded-r-md w-16 transition-colors duration-200
    lg:p-2
    dark:border-slate-500;
  }
</style>