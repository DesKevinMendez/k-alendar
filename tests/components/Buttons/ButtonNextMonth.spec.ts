import ButtonNextMonth from '@/components/Buttons/ButtonNextMonth.vue';
import { mount, VueWrapper } from '@vue/test-utils';
import { DateTime } from 'luxon';
import { Date2024Oct23 } from '../../fakes/monthDays';

const dateToTest = '2024-02-10';
const mockGenerateCalendar = vi.fn().mockReturnValue(Date2024Oct23);
const currentDate = { value: DateTime.fromISO(dateToTest) };
const monthDays = { value: [] };

vi.mock('@/composables/useRenderCalendar', () => ({
  default: vi.fn(() => ({
    currentDate,
    monthDays,
    generateCalendar: mockGenerateCalendar,
  })),
}));

let wrapper: VueWrapper<InstanceType<typeof ButtonNextMonth>>

describe('ButtonNextMonth', () => {

  beforeEach(() => {
    wrapper = mount(ButtonNextMonth);
  });

  afterEach(() => {
    vi.clearAllMocks()
  })

  it('should render the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('init values for composable useRenderCalendar', () => {
    expect(monthDays.value).toStrictEqual([]);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO(dateToTest));
  })

  it('do click on the button', async () => {
    const expected = '2024-03-10'

    await wrapper.find('button').trigger('click');

    expect(monthDays.value).toStrictEqual(Date2024Oct23);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO(expected));
    expect(mockGenerateCalendar).toHaveBeenCalledWith(expected);
    expect(wrapper.emitted('handle')).toStrictEqual([[expected]]);

    await wrapper.find('button').trigger('click');
    expect(wrapper.emitted('handle')).toStrictEqual([[expected], ['2024-04-10']]);
    expect(currentDate.value).toStrictEqual(DateTime.fromISO('2024-04-10'));
  })
});