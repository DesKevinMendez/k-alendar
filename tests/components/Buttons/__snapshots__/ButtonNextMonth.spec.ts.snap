// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonNextMonth > should render the component 1`] = `"<button data-v-f3d09f65="" type="button" class="k-alendar-navegation-left">â†’</button>"`;
